src/core/SkPicture.cpp

```diff
@@ -49,11 +49,11 @@ SkPicture::SkPicture() {
 
 SkPicture::~SkPicture() {
     if (fAddedToCache.load()) {
-        if (!SkRecordReplayIsRecordingOrReplaying() || !SkRecordReplayAreEventsDisallowed()) {
+        if (!SkRecordReplayIsRecordingOrReplaying(/* "leak-references" */) || !SkRecordReplayAreEventsDisallowed()) {
```

Comment:
Rather than having a feature test here it might be better to update SkRecordReplayAreEventsDisallowed so it checks the disallow-events feature like v8::recordreplay::AreEventsDisallowed does.

BTW the difficulties here in dealing with features in places that don't have access to V8 is a pretty strong argument in favor of putting the feature mechanism inside the recorder so it can be used from anywhere.

Reply:
1. Why not check against `"leak-references"` as well? That's what we would usually do, no?
2. I added a Linear issue to address the feature check situation. Since that's a bigger follow-up issue, I will merge this as-is, for now. Hope that's Ok,

Reply:
Linear issue: https://linear.app/replay/issue/RUN-1898/add-featureenabled-to-recorder-api